docs/notes/2.23.x.md

```diff
@@ -162,78 +149,53 @@ exclude = ["**/tests/*"]
-- To make this easier, the `python_google_cloud_function` target will now automatically select a `complete_platforms` based on the value of the `runtime` field, similar to the AWS targets.
-- The `complete_platforms` field may now be specified as either a `file` target or a `resource` target.
-- The `python_aws_lambda_function` and `python_aws_lambda_layer` targets now allow specifying the machine architecture to use for the Lambda function via [the new `architecture` field](https://www.pantsbuild.org/2.23/reference/targets/python_aws_lambda_function#architecture). Additionally, both targets now support automatically selecting the `complete_platforms` used for building PEX files based on the values of the `architecture` and `runtime` fields together.
+Changed some potentially confusing behavior in how outputs from `adhoc_tool` and `shell_command` running in an `experimental_workspace_environment` are found. Previously, for most invoked processes, Pants interpreted `output_files` and `output_directories` as relative paths relative to the configured working directory from the `workdir` field. In-workspace executions will now capture from the root of the temporary sandbox directory used during execution and will not prepend the working directory (which still applies to the invoked process). The `output_files` and `output_directories` fields continue to be interpreted relative to the working directory for executions in all other environments (e.g., local, remote, and Docker).
 
-Lockfile generation now respects hermetic python selection. That means that lockfile generation can use an interpreter provided by the [`pants.backend.python.providers.experimental.pyenv` backend](https://www.pantsbuild.org/2.23/reference/subsystems/pyenv-python-provider).
+[Added `workspace_invalidation_sources` field](https://www.pantsbuild.org/2.23/reference/targets/adhoc_tool#workspace_invalidation_sources) to `adhoc_tool` and `shell_command` target types. This new field allows declaring that these targets depend on files without bringing those files into the execution sandbox, but that the target should still be re-executed if those files change. This is intended to work with the `workspace_environment` support where processes are executed in the workspace and not in a separate sandbox.
 
-A new `uv_requirements` macro has been added to allow importing [development dependencies specified in `pyproject.toml` files under the `[tool.uv]` section](https://docs.astral.sh/uv/concepts/dependencies/#development-dependencies).
+#### Docker
 
-#### Terraform
+Docker inference is improved. Pants can now make inferences by target address for targets supporting `pants package`, and `file` targets can be included by filename. See the [documentation on Docker dependency inference](https://www.pantsbuild.org/2.23/docs/docker#dependency-inference-support) for details
```

Comment:
I didn't know this!  Maybe this is a highlight?

Reply:
Added!